ecl/hqlcpp/hqlresource.cpp

```diff
@@ -2552,6 +2552,7 @@ void SpillerInfo::addSpillFlags(HqlExprArray & args, bool isRead)
         if (!original->isDataset() || hasSingleRow(original))
             args.append(*createAttribute(rowAtom));                 // add a flag so the selectnth[1] can be removed later...
     }
+    args.append(*createExprAttribute(_signed_Atom, createConstant(true)));
```

Comment:
I think this should possibly move out of addSpillFlags to line 2654 in the same file, otherwise it gets added to reads and writes of temporary results, and writes of spill files.